app/models/order.rb

```diff
@@ -1,21 +1,25 @@
+# Order model
 class Order < ApplicationRecord
-  belongs_to :order_status
+  belongs_to :user
   has_many :order_items
-  before_create :set_order_status
-  before_save :pdate_subtotal
+  before_save :update_subtotal
+  accepts_nested_attributes_for :order_items
 
   def subtotal
-    order_items.collect { |oi| oi.valid? ? (oi.quantity * oi.unit_price) : 0 }.sum
+    order_items.collect { |o| o.valid? ? (o.quantity * o.unit_price) : 0 }.sum
   end
 
-  private
+  def total
+    shipping + subtotal
+  end
 
-  def set_order_status
-    self.order_status_id = 1
+  def shipping
+    10 / 100 * subtotal
```

Comment:
you need `100.to_f` I think..otherwise this will always return 0. 10 / 100 returns 0 and 10 / 100.to_f returns 0.1